buildpack/telemetry/metrics.py

```diff
@@ -186,21 +123,119 @@ def configure_metrics_registry(m2ee):
     ):
-        paidapps_registries.append(STATSD_REGISTRY)
+        paidapps_registries.append(get_statsd_registry())
 
     return paidapps_registries
 
 
+def get_influx_registry():
+    # Runtime configuration for influx registry
+    # This enables the new stream of metrics coming from micrometer instead
+    # of the admin port.
+    # https://docs.mendix.com/refguide/metrics#registries-configuration
+    # NOTE: Metrics are usually dot separated. But each registry has its
+    # own naming format. For instance, a metric like
+    # `a.name.like.this` would appear as `a_name_like_this` in
+    # influx-formatted metrics output. Hence the filter names uses the
+    # dot-separated metric names.
+    return {
+        "type": "influx",
+        "settings": {
+            "uri": "http://localhost:8086",
+            "db": "mendix",
+            "step": "10s",
+        },
+        "filters": [
+            # Login metrics needs to be enabled explicitly as it's disabled
+            # by default
+            {
+                "type": "nameStartsWith",
+                "result": "accept",
+                "values": ["mx.runtime.user.login"],
+            },
+            # Filter out irrelevant metrics to reduce
+            # the payload size passed to TSS/TFR
+            # https://docs.mendix.com/refguide/metrics#filters
+            {
+                "type": "nameStartsWith",
+                "result": "deny",
+                "values": ["commons.pool", "jvm.buffer"],
+            },
+        ],
+    }
+
+
+def get_statsd_registry():
+    return {
+        "type": "statsd",
+        "settings": {"port": datadog.get_statsd_port()},
+        "filters": [
+            {
+                "type": "nameStartsWith",
+                "result": "accept",
+                "values": get_apm_metrics_filter("APM_METRICS_FILTER_ALLOW"),
+            },
+            {
+                "type": "nameStartsWith",
+                "result": "deny",
+                "values": get_apm_metrics_filter("APM_METRICS_FILTER_DENY"),
+            },
+        ],
+    }
+
+
+def get_apm_metrics_filter(env_var_name):
+    """
+    If we use empty string ("") in the filters that we use for statsd registry,
+    it accepts/denies every metric since we use type as `nameStartsWith`.
+    To prevent breaking the functionality because of this, we need to make sure
+    that we pass empty string to the registry filters only if it's intentional.
+    So, we strip the leading and trailing commas. Additionally we remove all
+    the white spaces to prevent any unintentional mistakes.
+    """
+    env_var_value = os.getenv(env_var_name)
+    if env_var_value is None:
+        return []
+    apm_filter = env_var_value.replace(" ", "").strip(",").split(",")
+    logging.info("APM metrics filter for %s is: %s", env_var_value, apm_filter)
+    return apm_filter
+
+
+def get_freeapps_registry():
+    # For freeapps we push only the session & login metrics
+    return [
```

Comment:
I know you didn't change that, but I would like to suggest that we return a dictionary here and add the list in the method that calls it, as it's done for the others.

Reply:
updated, thanks for the suggestion 👍